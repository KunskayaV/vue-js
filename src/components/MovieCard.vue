
<template>
  <figure class="poster">
    <img class="poster-image" :src="item.posterUrl" v-lazyload :alt="item.shortDescription" width="300" height="400" />
    <figcaption class="info">
      <div class="info-wrapper">
        <span class="name">{{  item.name }}</span>
        <span class="genre">{{  item.genre }}</span>
      </div>
      <ItemReleaseDateVue :date="item.date" />
    </figcaption>
  </figure>
</template>


<script setup lang="ts">
import type { TMovie } from '@/types';
import ItemReleaseDateVue from './ItemReleaseDate.vue';
import type { Directive } from 'vue';


defineProps<{ item: TMovie; }>();

const vLazyload: Directive<HTMLImageElement> = {
  mounted: (el) => el.setAttribute('loading', "lazy"),
}

</script>

<style scoped>
  .poster {
    display: flex;
    flex-direction: column;
    width: 300px;
    height: 400px;
  }

  .poster-image {
    background-color: #bdbdbd;
  }

  .info {
    margin-top: 12px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
    color: #bdbdbd;
  }

  .info-wrapper {
    display: flex;
    flex-direction: column;
  }

  .name {
    font-size: 1.3rem;
    font-weight: 600;
    letter-spacing: 1.1px;
  }

  .genre {
    margin-top: 4px;
    font-size: 0.8rem;
    font-weight: 500;
  }

</style>