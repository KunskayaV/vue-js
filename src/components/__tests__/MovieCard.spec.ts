import { describe, it, expect, beforeEach, vi } from 'vitest'

import { mount } from '@vue/test-utils'
import MovieCard from '@/components/MovieCard.vue'
import { mockMovie } from '@/__mocks__'

describe('MovieCard', () => {
  beforeEach(() => {
    vi.spyOn(Math, 'random').mockReturnValue(1)
  })

  it('should match snapshot', () => {
    const wrapper = mount(MovieCard, { props: { item: mockMovie } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correct title', () => {
    const wrapper = mount(MovieCard, { props: { item: mockMovie } })

    expect(wrapper.find('[date-testid=title]').text()).toBe(mockMovie.title)
  })

  it('should render correct genre', () => {
    const wrapper = mount(MovieCard, { props: { item: mockMovie } })

    expect(wrapper.find('[date-testid=genre]').text()).toBe(
      mockMovie.genres[0] + ' & ' + mockMovie.genres[1]
    )
  })
})
