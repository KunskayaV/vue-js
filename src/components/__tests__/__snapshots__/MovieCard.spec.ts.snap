// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard > should match snapshot 1`] = `
"<figure data-v-8019de98=\\"\\" class=\\"poster\\"><img data-v-8019de98=\\"\\" class=\\"poster-image\\" src=\\"https://loremflickr.com/g/320/240/poster,movie,film?random=567&amp;lock=132\\" alt=\\"Chronicles the experiences of a formerly successful banker as a prisoner in the gloomy jailhouse of Shawshank after being found guilty of a crime he did not commit. The film portrays the man's unique way of dealing with his new, torturous life; along the way he befriends a number of fellow prisoners, most notably a wise long-term inmate named Red.                Written by
J-S-Golden\\" width=\\"300\\" height=\\"400\\" loading=\\"lazy\\">
  <figcaption data-v-8019de98=\\"\\" class=\\"info\\">
    <div data-v-8019de98=\\"\\" class=\\"info-wrapper\\"><span data-v-8019de98=\\"\\" class=\\"name\\" date-testid=\\"title\\">The Shawshank Redemption</span><span data-v-8019de98=\\"\\" class=\\"genre\\" date-testid=\\"genre\\">Crime &amp; Drama</span></div><span data-v-3f3ba9c4=\\"\\" data-v-8019de98=\\"\\" class=\\"item-release-date\\">1995</span>
  </figcaption>
</figure>"
`;
